staging/src/k8s.io/component-helpers/nodedeclaredfeatures/features/inplacepodresize/pod_level_resource_resize.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inplacepodresize
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/component-helpers/nodedeclaredfeatures"
+)
+
+// Ensure the feature struct implements the unified Feature interface.
+var _ nodedeclaredfeatures.Feature = &podLevelResourcesResizeFeature{}
+
+const (
+	// IPPRPodLevelResourcesFeatureGate is the feature gate for IPPRPodLevelResourcesVerticalScalingFeatureGate.
+	IPPRPodLevelResourcesFeatureGate = "InPlacePodLevelResourcesVerticalScaling"
+	// PodLevelResourcesResize is a declared feature that indicates a node supports in-place pod resize for pod level resources.
+	PodLevelResourcesResize = "PodLevelResourcesResize"
+)
+
+// Feature is the implementation of the `PodLevelResourcesResize` feature.
+var PodLevelResourcesResizeFeature = &podLevelResourcesResizeFeature{}
+
+type podLevelResourcesResizeFeature struct{}
+
+func (f *podLevelResourcesResizeFeature) Name() string {
+	return PodLevelResourcesResize
```

Comment:
Since this one is tightly coupled to the feature gate, I think you should just use the feature gate name here.